lms/static/scripts/frontend_apps/utils/url.ts

```diff
@@ -0,0 +1,33 @@
+/**
+ * Replace parameters in a URL template with values from a `params` object and
+ * returns the expanded URL.
+ *
+ *   replaceURLParams('/things/:id', {id: 'foo'}) => '/things/foo'
+ *
+ * @throws Error in case any provided param does not have a matching
+ *               placeholder in the template URL
+ */
+export function replaceURLParams<Param>(
+  urlTemplate: string,
+  params: Record<string, Param>,
+): string {
+  const paramEntries = Object.entries(params);
+  let url = urlTemplate;
+
+  paramEntries.forEach(([param, value]) => {
+    const urlParam = `:${param}`;
+    if (!url.includes(urlParam)) {
+      throw new Error(
+        `Parameter "${param}" not found in "${urlTemplate}" URL template`,
+      );
+    }
+
+    // Replace all occurrences of the same param in the template
+    url = url.replace(
+      new RegExp(urlParam, 'g'),
```

Comment:
This will have unexpected behavior if `urlParam` contains special characters. You can use [`String.prototype.replaceAll`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replaceAll#browser_compatibility) instead.

Reply:
I didn't use `replaceAll` because my IDE complained about it. I think it's just because we have `"lib": ["es2018", "dom"],` in the tsconfig, and it should work with `es2021`.

Reply:
I have changed that in tsconfig. Now tsc does not complain.